### ISSUE TITLE
Webhook

### ISSUE THREAD

**Issue by elihun** [kotlin-telegram-bot/kotlin-telegram-bot#44]
Hello! Is there any way to use webhooks for update handling?

**Comment by seik** [comment]
Hi @ilya-gr, right now there is no option for webhooks. I can't say I'm going to implement it since it's something I'm not going to use by myself.

If someone wants to give a take on it the PRs are open! I'm going to leave this issue ope as a reference.

**Comment by ilya-gr** [comment]
Hi @seik, thank you for the answer!

**Comment by vjgarciag96** [comment]
I'm going to start working on this, stay tuned!

**Comment by vicmosin** [comment]
@vjgarciag96 Is there any update regarding the issue?

**Comment by vjgarciag96** [comment]
Hi @vicmosin!! I have a branch with a working version. However, I wanted to test that all the certificates stuff is working in a real environment, but I had no time to finish checking that. Do you want to contribute on this?. If so, I could upload the work in progress version of my branch to let you check how it's going and to also let you contributing to it if you want.

**Comment by vjgarciag96** [comment]
Here you can check the advances of my branch -> https://github.com/vjgarciag96/kotlin-telegram-bot/commit/7cb55145f2002e01e1d46a211adab9b7edd75670

If you want to contribute on it, please let me know (cc/ @vicmosin)

**Comment by vicmosin** [comment]
@vjgarciag96 Could you please put all the changes into your branch and describe a bit what's still missing? I think I could check it

**Comment by vjgarciag96** [comment]
Sure, the branch is now updated with the latest changes!

Regarding the changes of the branch...

I have basically added the three API operations needed to work with webhooks:

- `setWebhook`. This is the one that has more complexity. Well, the only "complex" point is that the certificate file can be sent in different ways (using a file id, an url of an already uploaded file or as a `multipart/form-data` chunk)
- `getWebhookInfo`
- `deleteWebhook`

As the size of the current API client was getting too big, I created another class to handle the`setWebhook` operation, which is a new collaborator of the current API client. The current API client just delegates the calls to `setWebhook` to it. Maybe we could find an abstraction for operations involving files and avoid this new class, but it's probably out of scope of this PR.

In addition, the `Bot` class has been slightly updated in order to support updates from either the webhook or the `getUpdates` method (previous implementation was not ready for that).

Finally, a new module has been added with a sample bot using webhooks that is run within a `Netty` server with `ktor` framework.

The missing things are basically reviewing the code (and discussing enhancements, suggestions, whatever) and checking the certificates stuff is working in a real environment. This would involve deploying the bot in any server and setting up a webhook for it using the three different methods available in order to check that all work properly in a real environment.
